docs/guides/permissions.md

```diff
@@ -26,11 +28,16 @@ class Query:
 ```
 
 Your `has_permission` method should do the work to check if this request
-has permission to the field. If the `has_permission` method returns `True` or
-a truthy value then the field access will go ahead. If the `has_permission`
-method returns `False` or a falsy value then an error will be raised using
-the `message` class attribute. See [Dealing with Errors](/docs/guides/errors)
-for more information on how errors are handled.
+has permission to the field.
+
+If the `has_permission` method returns a truthy value or an awaitable resolving to a
+truthy value then the field access will go ahead.
```

Comment:
I don't think you need to spell this out, it's enough explaining that the `has_permission` method can be an async method.